### PR TITLE
feat(controls): Add react versions of core media controls

### PR DESCRIPTION
**Pull request by jstoffan**

### CODE REVIEW

File: src/lib/viewers/controls/media/PlayPauseToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import noop from 'lodash/noop';
+import IconPlay24 from '../icons/IconPlay24';
+import IconPause24 from '../icons/IconPause24';
```
Comment (ConradJChan) [inline src/lib/viewers/controls/media/PlayPauseToggle.tsx:4]: 🔤

File: src/lib/viewers/controls/media/PlayPauseToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    isPlaying?: boolean;
+    onPlayPause: (isPlaying: boolean) => void;
+};
+
+export default function PlayPauseToggle({ isPlaying, onPlayPause = noop }: Props): JSX.Element {
+    const Icon = isPlaying ? IconPause24 : IconPlay24;
+    const title = isPlaying ? __('media_pause') : __('media_play');
+
+    return (
+        <MediaToggle className="bp-PlayPauseToggle" onClick={(): void => onPlayPause(!isPlaying)} title={title}>
```
Comment (ConradJChan) [inline src/lib/viewers/controls/media/PlayPauseToggle.tsx:18]: Is the return type on the `onClick` callback needed?

File: src/lib/viewers/controls/media/SettingsControls.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export default function SettingsControls(): JSX.Element {
+    const [isVisible, setVisible] = React.useState(false);
+
+    const handleClick = (): void => {
+        setVisible(!isVisible);
+    };
+
+    return (
+        <div className="bp-SettingsControls">
+            <MediaToggle className="bp-SettingsControls-toggle" onClick={handleClick} title={__('media_settings')}>
```
Comment (ConradJChan) [inline src/lib/viewers/controls/media/SettingsControls.tsx:15]: Do we need to add the little gear rotate animation when toggled?

**Review by ConradJChan — DISMISSED** [review]

File: src/lib/viewers/controls/media/PlayPauseToggle.tsx (hunk 2 above)
Comment (jstoffan) [reply src/lib/viewers/controls/media/PlayPauseToggle.tsx:18]: Yeah, every function needs an explicit return type per our current TS config.

File: src/lib/viewers/controls/media/SettingsControls.tsx (hunk 3 above)
Comment (jstoffan) [reply src/lib/viewers/controls/media/SettingsControls.tsx:15]: Eventually, but this is mainly a placeholder for now.

**Review by ConradJChan — APPROVED** [review]

**Pull request merged**